server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -714,6 +743,84 @@ private RemapGlobalOrds(CardinalityUpperBound cardinality) {
             bucketOrds = LongKeyedBucketOrds.build(context.bigArrays(), cardinality);
         }
 
+        @Override
+        String describe() {
+            return "remapStarTree";
+        }
+
+        @Override
+        void collectDebugInfo(BiConsumer<String, Object> add) {
+            add.accept("total_buckets", bucketOrds.size());
+        }
+
+        @Override
+        void globalOrdsReady(SortedSetDocValues globalOrds) {}
+
+        @Override
+        void collectGlobalOrd(long owningBucketOrd, int doc, long globalOrd, LeafBucketCollector sub) throws IOException {
+            long bucketOrd = bucketOrds.add(owningBucketOrd, globalOrd);
+            if (bucketOrd < 0) {
+                bucketOrd = -1 - bucketOrd;
+                collectExistingBucket(sub, doc, bucketOrd);
+            } else {
+                collectBucket(sub, doc, bucketOrd);
+            }
+        }
+
+        @Override
+        long globalOrdToBucketOrd(long owningBucketOrd, long globalOrd) {
+            return bucketOrds.find(owningBucketOrd, globalOrd);
+        }
+
+        @Override
+        void forEach(long owningBucketOrd, BucketInfoConsumer consumer) throws IOException {
+            if (bucketCountThresholds.getMinDocCount() == 0) {
+                for (long globalOrd = 0; globalOrd < valueCount; globalOrd++) {
+                    if (false == acceptedGlobalOrdinals.test(globalOrd)) {
+                        continue;
+                    }
+                    /*
+                     * Use `add` instead of `find` here to assign an ordinal
+                     * even if the global ord wasn't found so we can build
+                     * sub-aggregations without trouble even though we haven't
+                     * hit any documents for them. This is wasteful, but
+                     * settings minDocCount == 0 is wasteful in general.....
+                     */
+                    long bucketOrd = bucketOrds.add(owningBucketOrd, globalOrd);
+                    long docCount;
+                    if (bucketOrd < 0) {
+                        bucketOrd = -1 - bucketOrd;
+                        docCount = bucketDocCount(bucketOrd);
+                    } else {
+                        docCount = 0;
+                    }
+                    consumer.accept(globalOrd, bucketOrd, docCount);
+                }
+            } else {
+                LongKeyedBucketOrds.BucketOrdsEnum ordsEnum = bucketOrds.ordsEnum(owningBucketOrd);
+                while (ordsEnum.next()) {
+                    if (false == acceptedGlobalOrdinals.test(ordsEnum.value())) {
+                        continue;
+                    }
+                    consumer.accept(ordsEnum.value(), ordsEnum.ord(), bucketDocCount(ordsEnum.ord()));
+                }
+            }
+        }
+
+        @Override
+        public void close() {
+            bucketOrds.close();
+        }
+    }
+
+    private class RemapGlobalOrdsStarTree extends CollectionStrategy {
```

Comment:
Lets extend RemapGlobalOrds and just add `collectGlobalOrdST` method, lets also rename to `collectGlobalOrdsForStarTree`